### PR TITLE
perf: internal objects optimization

### PR DESCRIPTION
**Pull request by iproudhon**
For internal account, balance, supply objects, use protobuf marshal / unmarshal instead of much slower MarshalInterface. Also uses []byte for addresses instead of string in the above objects and MsgSend for performance as well.

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: x/bank/keeper/msg_server.go [hunk 1]
```diff
@@ -29,20 +29,14 @@ func (k msgServer) Send(goCtx context.Context, msg *types.MsgSend) (*types.MsgSe
...
 	}
-	from, err := sdk.AccAddressFromBech32(msg.FromAddress)
-	if err != nil {
-		return nil, err
-	}
-	to, err := sdk.AccAddressFromBech32(msg.ToAddress)
-	if err != nil {
-		return nil, err
-	}
+	from := sdk.AccAddress(msg.FromAddress)
```
Comment (brew0722) [inline x/bank/keeper/msg_server.go:32]: How does it work exist prefix validation by bech32?('link...')

Comment (egonspace) [reply x/bank/keeper/msg_server.go:32]: Client should assign binary address value to `Address` type of all messages.
We will have to think about this and discuss it.

Comment (brew0722) [reply x/bank/keeper/msg_server.go:32]: Ah, I temporally wrong things thought that the address bytes contained bech32 prefix bytes. I understood now.

**Comment by iproudhon** [comment]
Further change is needed.

**Pull request closed without merging**